metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1416,6 +1424,29 @@ class MetalsLanguageServer(
         scribe.debug(s"Executing SuperMethodHierarchy ${command}")
         supermethods.jumpToSelectedSuperMethod(params).asJavaObject
 
+      case ServerCommands.ResetChoicePopup() =>
+        val argsMaybe = Option(params.getArguments())
+        (argsMaybe
+          .map { args =>
+            args.asScala.headOption match {
+              case Some(argObject: JsonPrimitive) =>
+                val arg = argObject.asInstanceOf[JsonPrimitive]
+                val value = arg.getAsString().replace("+", " ")
+                scribe.debug(
+                  s"Executing ResetChoicePopup ${command} for choice ${value}"
+                )
+                popupChoiceReset.reset(value)
+
+              case _ =>
+                scribe.debug(
+                  s"Executing ResetChoicePopup ${command} in interactive mode."
+                )
+                popupChoiceReset.interactiveReset()
+            }
+          })
+          .getOrElse(Future.successful(()))
```

Comment:
So not a big deal, but what do you think about instead of just returning a `Future.successful()` here if there are no arguments, to instead do the `interactiveReset()`. Main reason being right now for clients that won't ever send in params, they have to register the command separately because it _has_ to be registerd with the arguments field of an empty array. So for `coc-metals` for example, that means I can't just add a one-liner here:
https://github.com/scalameta/coc-metals/blob/630450a7cd1e777ddf1392e672838de2e2965b7c/src/index.ts#L248-L260

and instead I'd need to do this:

```javascript
    registerCommand("metals.reset-choice-interactive", () => {
      client.sendRequest(ExecuteCommandRequest.type, {
        command: "reset-choice",
        arguments: [],
      });
    });
```
Again, it's a nitpick, but for me having to set arguments to an empty array all the time doesn't make sense. So it might be nice that if the server gets the `reset-choice` request with no params, it still works.

Reply:
Definitely worth changing it as you suggested :+1:

Reply:
changed